### PR TITLE
Material Design Checkbox

### PR DESCRIPTION
**Pull request by TanayParikh**
- Implemented a Material Design checkbox

### CODE REVIEW

File: src/foam/u2/CheckBox.js [hunk 1]
```diff
@@ -66,11 +66,11 @@ foam.CLASS({
         margin: 8px;
         padding: 8px;
       }
-
+
```
Comment (kgrgreer) [inline src/foam/u2/CheckBox.js]: Remove tab.

File: src/foam/u2/CheckBox.js [hunk 2]
```diff
@@ -66,11 +66,11 @@ foam.CLASS({
         margin: 8px;
         padding: 8px;
       }
-
+      
       ^label {
         color: #444;
         flex-grow: 1;
-        margin-right: 12px;
+        margin-left: 12px;
```
Comment (kgrgreer) [inline src/foam/u2/CheckBox.js:73]: Does this change the appearance for any current CheckBox users?

File: src/foam/u2/md/OverlayDropdown.js [hunk 3]
```diff
@@ -150,8 +150,8 @@ foam.CLASS({
       
       if ( Number.isNaN(margin) ) margin = 0;
 
-      return Math.min(border + last.offsetTop + last.offsetHeight + margin,
-          this.window.innerHeight - this.dropdownE_.el().getBoundingClientRect().top);
+      return (Math.min(border + last.offsetTop + last.offsetHeight + margin,
```
Comment (kgrgreer) [inline src/foam/u2/md/OverlayDropdown.js]: Outer parens not needed.

File: src/foam/u2/md/OverlayDropdown.js [hunk 4]
```diff
@@ -150,8 +150,8 @@ foam.CLASS({
...
+          this.window.innerHeight - this.dropdownE_.el().getBoundingClientRect().top)) - 13;
```
Comment (kgrgreer) [inline src/foam/u2/md/OverlayDropdown.js]: How about a comment explaining where the 13 comes from, or make it a constant.

File: src/foam/u2/CheckBox.js (hunk 2 above)
Comment (TanayParikh) [reply src/foam/u2/CheckBox.js:73]: It should've been `margin-left` from the start, adding a margin to the right doesn't do anything as the checkbox is to the left. (I added it in last month so probably not too many users of this just yet).

File: src/foam/u2/md/OverlayDropdown.js (hunk 4 above)
Comment (TanayParikh) [reply src/foam/u2/md/OverlayDropdown.js]: Made it a constant

File: src/foam/u2/md/OverlayDropdown.js (hunk 3 above)
Comment (TanayParikh) [reply src/foam/u2/md/OverlayDropdown.js]: Removed

File: src/foam/u2/CheckBox.js (hunk 1 above)
Comment (TanayParikh) [reply src/foam/u2/CheckBox.js]: Removed

File: src/foam/u2/md/OverlayDropdown.js [hunk 5]
```diff
@@ -83,6 +83,11 @@ foam.CLASS({
       value: 0
     },
     {
+      class: 'Int',
```
Comment (kgrgreer) [inline src/foam/u2/md/OverlayDropdown.js]: Is this going to change? If not, put in constants: as BOTTOM_OFFSET.

Comment (TanayParikh) [reply src/foam/u2/md/OverlayDropdown.js]: Oh got it, updated.

**Pull request merged**